ff-api-update.php

```diff
@@ -1,32 +1,73 @@
 <?php
- 
+
 /* 
  * refresh the number of active nodes in the freifunk fulda community.
  * the node information is gathered from the ffmap nodes.json file and
  * written to the freifunk.net API file.
  */
- 
-$NODELIST = 'https://meshviewer.fulda.freifunk.net/data/nodes.json';
-$APIFILE = '/path/to/FreifunkFulda-api.json';
- 
-$map_json = file_get_contents($NODELIST);
+
+# disable SSL verification to prevent errors
+$arrContextOptions=array(
+    "ssl"=>array(
+        "verify_peer"=>false,
+        "verify_peer_name"=>false,
+    ),
+);
+
+# json validation
+function validateJson($json) {
+	$json_test = json_decode($json);
+	if (json_last_error() !== JSON_ERROR_NONE || $json_test === null) {
+	  die "Contains invalid json data";
+	}
+}
+
+# include configs
+require (__DIR__ . '/config.php');
+
+# Download, verify and decode directory.json
+$list_json = file_get_contents($APILISTFILE, false, stream_context_create($arrContextOptions));
+validateJson($list_json);
+$list_php = json_decode($list_json);
+
+# Select community API link from directory.json
+$apifile = $list_php->{$COMMUNITY};
+
+# Download, verify and decode nodes list
+$map_json = file_get_contents($NODELIST, false, stream_context_create($arrContextOptions));
+validateJson($map_json);
 $map_php = json_decode($map_json);
- 
+
+# Count all nodes
 $nodes = count((array)($map_php->{'nodes'}));
- 
+
+# Remove gateways and offline nodes
```

Comment:
If you loop over the nodes anyways, you can increment for all online nodes and avoid the call to count above.